tests/unit/payment-methods/test-class-upe-payment-gateway.php

```diff
@@ -536,17 +663,43 @@ public function test_create_payment_intent_with_automatic_capture() {
 		$order_id = $order->get_id();
 		$intent   = WC_Helper_Intention::create_intention( [ 'status' => 'requires_payment_method' ] );
 		$this->mock_upe_gateway->settings['manual_capture'] = 'no';
-		$this->mock_api_client
-			->expects( $this->once() )
-			->method( 'create_intention' )
+		$request = $this->mock_wcpay_request( Create_Intention::class );
+		$request->expects( $this->once() )
+			->method( 'set_amount' )
+			->with( $intent->get_amount() );
+
+		$request->expects( $this->once() )
+			->method( 'set_currency_code' )
+			->with( strtolower( $intent->get_currency() ) );
+
+		$request->expects( $this->once() )
+			->method( 'set_capture_method' )
+			->with( false );
+
+		$request->expects( $this->once() )
```

Comment:
Is there a reason to remove it? This method is called with this parameter, and there is noting bad in mocking that. I removed that from other tests, but it would be nice to have it at least somewhere to validate is it called or setter correctly.

Reply:
Same here, having it once here and there it's fine, there was just no need to test it everywhere.